src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [];
+
+  for (const action of actions) {
+    const currentState = {};
+
+    if (stateClones.length !== 0) {
+      Object.assign(currentState, stateClones[stateClones.length - 1]);
+    }
+
+    if (stateClones.length === 0) {
+      Object.assign(currentState, state);
+    }
+
+    switch (action.type) {
```

Comment:
Use a destructurization here for `action`
```suggestion
   const {} = action;
   
    switch (type) {
```